src/index.js

```diff
@@ -0,0 +1,748 @@
+    }
+
+    const { fileId, mediaId } = data;
+    let url = mediaId ? `api/v4/media/${mediaId}/save/` : 'api/v4/media/save/';
+
+    if (fileId) {
+      url += `${fileId}/`;
+    }
+
+    return this.api.send('POST', url, data);
+  }
+
+  /**
+   * Uploads arbirtrarily sized buffer or stream file to provided S3 endpoint in chunks and registers each chunk to Bynder.
+   * Resolves the passed init result and final chunk number.
+   * @async
+   * @param {Object} file An object containing the id of the desired collection.
+   * @param {String} file.filename The file name of the file to be saved
+   * @param {Buffer|Readable} file.body The file to be uploaded. Can be either buffer or a read stream.
+   * @param {Number} file.length The length of the file to be uploaded
+   * @param {String} fileId Unique file identifier
+   * @param {Number} size File byte size
+   * @return {Promisee<number>} Total number of chunks uploaded to the upload payload
+   */
+  async uploadFileInChunks(file, fileId, size) {
+    const { body } = file;
+    // Developers can use this to track file upload progress
+    this._chunks = Math.ceil(size / FILE_CHUNK_SIZE);
+    this._chunkNumber = 0;
+
+    // Iterate over the chunks and send them
+    while (this._chunkNumber <= (this._chunks - 1)) {
+      const start = this._chunkNumber * FILE_CHUNK_SIZE;
+      const end = Math.min(start + FILE_CHUNK_SIZE, size);
+      const chunk = body.slice(start, end);
+      const sha256 = create256HexHash(chunk);
+
+      await this.api.send('POST', `v7/file_cmds/upload/${fileId}/chunk/${this._chunkNumber}`, chunk, {
+        additionalHeaders: {
+          'Content-SHA256': sha256
+        }
+      })
+        .catch(error => {
+          throw new Error(`[${error.message} - ${error.status}] Chunk ${this._chunkNumber} not uploaded: ${error.body}`, error);
+        });
+
+      this._chunkNumber++;
+    }
+
+    return this._chunks;
+  }
+
+  /**
+   * Uploads an arbitrarily sized buffer or stream file and returns the uploaded asset information
+   * @see {@link https://bynder.docs.apiary.io/#reference/upload-assets}
+   * @async
+   * @param {Object} file An object containing the id of the desired collection.
+   * @param {String} file.filename The file name of the file to be saved
+   * @param {Buffer|Readable} file.body The file to be uploaded. Can be either buffer or a read stream.
+   * @param {Number} file.length The length of the file to be uploaded
+   * @param {Object} file.data An object containing the assets' attributes
+   * @param {String} fileId Optional file UUID.V4 identifier
+   * @return {Promise<object>} The information of the uploaded file, including IDs and all final file urls.
+   */
+  async uploadFile(file) {
+    const { body, filename, data } = file;
+    const { brandId } = data;
+    const bodyType = bodyTypes.get(body);
+    const size = getLength(file);
+
+    if (!brandId) {
+      return rejectValidation('upload', 'brandId');
```

Comment:
We don't need a brandId if we save to an existing asset (so if we go this route: https://bynder.docs.apiary.io/#reference/upload-assets/6-saving-a-completed-upload/save-as-a-new-asset-version )